fix: bug when creating CE from LR

gurneyalex opened the pull request:
if no consignee was set on the LR (possible if LR is Cost Estimate only)
then creating the CE would crash because we were wrongly using the consignee as
partner when playing the sale.order onchanges

gurneyalex committed:
fix: bug when creating CE from LR

if no consignee was set on the LR (possible if LR is Cost Estimate only)
then creating the CE would crash because we were wrongly using the consignee as
partner when playing the sale.order onchanges

gurneyalex committed:
change requirement rule for consignee_id

we only want consignee_id to be mandatory on a SO if the SO is *not*
cost_estimate_only.

gurneyalex committed:
fix CE creation

propagate cost_estimate_only from LR to CE

lepistone commented on logistic_requisition/test/requisition_create_cost_estimate_only.yml:
```diff
@@ -0,0 +1,123 @@
+            "product_id should be %s, received %s" %
+            (source.proposed_product_id, sale_line.product_id))
+
+        assert sale_line.name == line.description, (
```
I know that you know, but are you sure you don't want to import `nose.tools` assertions? I doubt it gets more little :cat: killed. (not blocking over that)

lepistone commented:
Code review OK with a remark on assertions. Once green, :+1:

lepistone committed:
remove print statement

lepistone commented:
I removed a `print` statement found by pylint.

lepistone committed:
lint

lepistone commented:
There is a new test failure related to something I worked on on another PR. I am investigating that.

nbessi commented:
:+1:  on principe but I'surprised that change do not impact more code especially the one run when
confirming the CE

lepistone committed:
adapt tests in logistic_budget to new logic

The test failure was difficult to understand. Maybe if that happens
again I'll consider using newids instead of mocks here. The ideal would
be to find the way to decouple the tests even more, so that that
logistic_budget really tests only budgets.

lepistone commented:
I'm not entirely happy about long stories in the tests, but it's good to have those anyway, and we'll be able  to improve them later.

Thanks!
